lib/Conversion/ConversionProvider.php

```diff
@@ -0,0 +1,133 @@
+		'documents' => [
+			'application/msword',
+			'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+			'application/vnd.openxmlformats-officedocument.wordprocessingml.template',
+			'application/vnd.oasis.opendocument.text-template',
+			'application/vnd.oasis.opendocument.text',
+		],
+		'sheets' => [
+			'application/vnd.ms-excel',
+			'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+			'application/vnd.openxmlformats-officedocument.spreadsheetml.template',
+			'application/vnd.oasis.opendocument.spreadsheet',
+			'application/vnd.oasis.opendocument.spreadsheet-template',
+		],
+		'presentations' => [
+			'application/vnd.ms-powerpoint',
+			'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+			'application/vnd.openxmlformats-officedocument.presentationml.template',
+			'application/vnd.oasis.opendocument.presentation',
+			'application/vnd.oasis.opendocument.presentation-template',
+		],
+		'drawings' => [
+			'application/vnd.visio',
+			'application/vnd.oasis.opendocument.graphics',
+			'application/vnd.oasis.opendocument.graphics-template',
+		],
+	];
+
+	public function __construct(
+		private RemoteService $remoteService,
+		private LoggerInterface $logger,
+		IFactory $l10nFactory,
+	) {
+		$this->l10n = $l10nFactory->get('richdocuments');
+	}
+
+	public function getSupportedMimeTypes(): array {
+		$toPdf = array_merge(
+			[],
+			self::MIME_TYPES['documents'],
+			self::MIME_TYPES['sheets'],
+			self::MIME_TYPES['presentations'],
+			self::MIME_TYPES['drawings'],
+		);
+
+		return [...$this->getMimeProvidersFor($toPdf, 'application/pdf')];
+	}
+
+	public function convertFile(File $file, string $targetMimeType): mixed {
+		$targetFileExtension = $this->getExtensionForMimeType($targetMimeType);
+		if ($targetFileExtension === null) {
+			throw new \Exception($this->l10n->t(
+				'Unable to determine the proper file extension for %1$s',
+				[$targetMimeType]
+			));
+		}
+		
+		return $this->remoteService->convertFileTo($file, $targetFileExtension);
+	}
+
+	private function getMimeProvidersFor(array $inputMimeTypes, string $outputMimeType): array {
+		$outputMimeInfo = $this->getMimeInfoFor($outputMimeType);
+		if ($outputMimeInfo === null) {
+			$this->logger->error($this->l10n->t('Unable to fetch information on $s', [$outputMimeType]));
+			throw new \Exception();
+		}
+
+		$conversionMimeProviders = [];
+		foreach ($inputMimeTypes as $mimeType) {
+			$conversionMimeProviders[] = new ConversionMimeProvider(
+				$mimeType,
+				...$outputMimeInfo
+			);
+		}
+
+		return $conversionMimeProviders;
+	}
+
+	private function getMimeInfoFor(string $targetMimeType): ?array {
+		foreach ($this->getTargetMimeTypes() as $mimeType => $mimeInfo) {
+			if ($mimeType === $targetMimeType) {
+				return [
+					'to' => $mimeType,
+					...$mimeInfo,
+				];
+			}
+		}
+
+		return null;
+	}
+
+	private function getTargetMimeTypes(): array {
+		return [
+			'application/pdf' => [
```

Comment:
Can we also extend to convert between document formats (e.g. docx -> odt, odt -> docx, odt -> rtf) where feasible 

https://www.collaboraonline.com/document-conversion/ has all, we can pick the most common ones where you think it makes sense